server/controllers/userController.js

```diff
@@ -0,0 +1,40 @@
+import database from "../models/index";
+import {
+  comparePassword,
+  generateJWToken,
+  httpResponse
+} from "../helpers/index";
+
+/**
+ * @description Generate login access token for a user
+ *
+ * @param {Request}  req
+ * @param {httpResponse} res
+ * @param {function} next
+ * @returns {object} HttpResponse {statusCode:int, success:boolean, data:object, message:string}
+ */
+export const userLogin = async (req, res, next) => {
+  const { email, password } = req.body;
+  if (email && password) {
+    const user = await database["Users"].findOne({ where: { email } });
+    if (user) {
+      const foundPassword = user.get("password");
+      const matchePassword = await comparePassword(password, foundPassword);
+      if (matchePassword) {
+        const jwtoken = await generateJWToken(user.get("id"));
+        return httpResponse(res, {
+          statusCode: 200,
+          success: true,
+          message: "success",
+          data: { token: jwtoken }
+        });
+      }
+    }
+  }
+  return httpResponse(res, {
+    statusCode: 401,
+    success: false,
+    message: "Invalid login credential",
+    data: null
+  });
+};
```

Comment:
Please put this logic in a `try catch` to catch any other errors live server errors etc

Reply:
I think I support Okwukwe on this. There should be a try-catch here, but Benny will decide if it's ok as it is. @Ndifreke seem to have a reason for not wanting to add the try-catch, even though we have told him to add it for the sake of the rubustness of the application.

Reply:
@mrkingz @eokwukwe Adding try catch in a call that does not throw an exception directly I think is a bad practice. You can't include try catch in all your code just in case something breaks. Instead I think it is better to handle all exception in the source of the call that will throw exception **if** you have access to the call. In the above case only findOne method of User will throw an exception **only** if the database connection was not initialized successfully, in this case the exception should be caught at the database.sync() call. This is done already.

Reply:
I disagree with this, if you check the async..await documentation you will see they actually recommend you use try..catches when you are utilizing async awaits https://javascript.info/async-await https://codeburst.io/node-express-async-code-and-error-handling-121b1f0e44ba. 

Also, you can include it in all your code 😃.